### ISSUE TITLE
Any update on relasing the training code?

### ISSUE THREAD

**Issue by fabiozappo** [microsoft/UniCL#7]
Hi sir! 

Thank you for your work, it's really itneresting to see how a combining image and text with label supervision basically matches the same results of a pure image classification training! This has inspired me a lot and now I am trying to replicate your results from this table 

![image](https://user-images.githubusercontent.com/46443190/207013306-00830c85-4cf5-4e8f-9ee7-0ddb70463372.png)

I am trying to train on my custom dataset of 22k classes and 1M images but there i'm still seeing a clear gap between training with your loss and a classification head. Did you try to run the experiment of this table on a bigger dataset such as imagenet 22k?

Releasing the training code for getting those results would be very helpfull for everyone interested in training a classification problem with your framework!

**Comment by jwyang** [comment]
Hi, @fabiozappo , did you try to reproduce the numbers listed in the above table based on cifar-10, cifar-100 or imagenet-1k? It would be great if you can start with the replicating of results on these three datasets. 

Regarding the training code, I will release the training code for imagenet-1k soon. Please stay tunned.

**Comment by fabiozappo** [comment]
Hi @jwyang , just to be fair I am here to say that I found a bug in my code, this ended up in a not fair comparison between the classification and contrastive problem. I fixed it and the gap is way smaller, so for future readers I reccomend to give this approach a try as it worked quite well for my private dataset!

Looking forward for the training code release!

**Comment by linzhiqiu** [comment]
Is the training code released now?

**Comment by wqtwjt1996** [comment]
> Hi, @fabiozappo , did you try to reproduce the numbers listed in the above table based on cifar-10, cifar-100 or imagenet-1k? It would be great if you can start with the replicating of results on these three datasets.
> 
> Regarding the training code, I will release the training code for imagenet-1k soon. Please stay tunned.

Hello: could you tell me are the code and pre-trained weights gonna available soon? Thanks!
